src/Geofence/Location.cs

```diff
@@ -154,7 +156,8 @@ private Location GetNominatimAddress(string city, double lat, double lng, string
                     wc.Headers.Add("User-Agent", Strings.BotName);
                     var json = wc.DownloadString(url);
                     dynamic obj = JsonConvert.DeserializeObject(json);
-                    return new Location(Convert.ToString(obj.display_name), city ?? unknown, Convert.ToDouble(obj.lat), Convert.ToDouble(obj.lon));
+                    var location_string = Smart.Format(nominatimSchema, obj);
```

Comment:
Need to think about solution for an Exception case when nominatim API is missing args we care